File: PROGRAMMERS/전화번호목록.py
phone_book = ["119", "97674223", "1195524421"]
# phone_book = ["123","456","789"]
# phone_book = ["12","123","1235","567","88"]

def solution(phone_book):
    answer = True
    phone_book.sort()
    print(phone_book)
    for x in range(len(phone_book)):
        if phone_book[x].startswith(phone_book[x-1]):
            answer = False
    return answer



print(solution(phone_book))

File: PROGRAMMERS/JAVA/src/Lv_2/전화번호목록.java
package Lv_2;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class 전화번호목록 {
    public static boolean solution(String[] phone_book) {
        boolean answer = true;
        Map<String, String> map = new HashMap<String, String>();

        for(int i=0; i<phone_book.length; i++){
            map.put(phone_book[i], "number");
        }

        for(String cellNumber:phone_book){
            for(int i=0; i<cellNumber.length(); i++){
                if(map.containsKey(cellNumber.substring(0,i))){
                    answer = false;
                }
            }
        }
        return answer;
    }

    public static void main(String[] args) {
//        String[] phone_book = {"119", "97674223", "1195524421"};
//        String[] phone_book = {"123","456","789"};
        String[] phone_book = {"12","123","1235","567","88"};
        boolean result = solution(phone_book);
        System.out.println("결과는 = " + result);
    }
}
